modules/tensor_mechanics/src/materials/ADComputeFiniteStrain.C

```diff
@@ -205,11 +213,12 @@ ADComputeFiniteStrain::computeQpIncrements(ADRankTwoTensor & total_strain_increm
       const auto lambda2 = std::sqrt(e_value[1]);
       const auto lambda3 = std::sqrt(e_value[2]);
 
-      N1.vectorOuterProduct(e_vector.column(0), e_vector.column(0));
-      N2.vectorOuterProduct(e_vector.column(1), e_vector.column(1));
-      N3.vectorOuterProduct(e_vector.column(2), e_vector.column(2));
+      // outer product of a vector with itself is guaranteed to be symmetric
+      N1.vectorSelfOuterProduct(e_vector.column(0));
+      N2.vectorSelfOuterProduct(e_vector.column(1));
+      N3.vectorSelfOuterProduct(e_vector.column(2));
```

Comment:
Here, N1, N2, and N3 will be modified, right? I am surprised that this didn't lead to any exodiffs.

Reply:
that's ... insane. Good catch. I think I changed my mind on the signature at some point and didn't go back to fix this. 

I do prefer the static signature, because the current state of any tensor this would be called on otherwise is irrelevant for the result. And the static form allows a more compact use of the call if it occurs inside a larger expression (i.e. no need to explicitly create a temporary variable)

Reply:
I agree.

Reply:
There should have at least been a compiler warning about an unused return value. I must have missed that.